packages/neo-one-monitor/src/Reporter.js

```diff
@@ -0,0 +1,130 @@
+/* @flow */
+import { interval } from 'rxjs/observable/interval';
+import { mergeScan } from 'rxjs/operators';
+import type { Subscription } from 'rxjs/Subscription';
+
+import BrowserLogger, {
+  type CollectingLoggerLogOptions,
+} from './BrowserLogger';
+import BrowserMetricsFactory, {
+  type MetricCollection,
+} from './BrowserMetricsFactory';
+import { KNOWN_LABELS } from './MonitorBase';
+
+const MAX_BACKLOG = 1000;
+
+export type Report = {|
+  logs: Array<CollectingLoggerLogOptions>,
+  metrics: MetricCollection,
+|};
+
+export default class Reporter {
+  _endpoint: string;
+  _subscription: Subscription;
+
+  constructor({
+    logger,
+    metricsFactory,
+    timer,
+    endpoint,
+  }: {|
+    logger: BrowserLogger,
+    metricsFactory: BrowserMetricsFactory,
+    timer: number,
+    endpoint: string,
+  |}) {
+    this._endpoint = endpoint;
+    const requestCounter = metricsFactory.createCounter({
+      name: 'browserRequestCounter',
+      help: 'Counts fetch requests from Browser Reporter to endpoint',
+      labelNames: [KNOWN_LABELS.ERROR],
+    });
+    const emptyBackReport = {
+      logs: [],
+      metrics: {
+        counters: {},
+        histograms: {},
+      },
+    };
+
+    this._subscription = interval(timer)
+      .pipe(
+        mergeScan(
+          async backReport => {
+            const logs = logger.collect();
+            const metrics = metricsFactory.collect();
+            let report = { logs, metrics };
+
+            if (backReport != null) {
+              report = this._addBackReport({ logs, metrics }, backReport);
```

Comment:
`report = this._addBackReport(report, backReport);` (convince yourself why this is possible)